### PR TITLE
SUS-4302 | Remove trailing slash from domain cache query

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-4302

### CODE REVIEW

File: extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php [hunk 1]
```diff
@@ -252,7 +252,7 @@ public function execute() {
 			 * (domain), $this->mCityId is unknown (set to false in constructor)
 			 */
 			wfProfileIn( __METHOD__."-domaincache" );
-			$key = WikiFactory::getDomainKey( $this->mServerName . '/' . $this->langCode );
+			$key = WikiFactory::getDomainKey( rtrim( $this->mServerName . '/' . $this->langCode, '/' ) );
```
Comment (sqreek) [inline extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php:255]: why rtrim?

File: extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php [hunk 2]
```diff
@@ -360,7 +353,7 @@ public function execute() {
 				 * store value in cache
 				 */
 				$oMemc->set(
-					WikiFactory::getDomainKey( $this->mServerName . '/' . $this->langCode ),
+					WikiFactory::getDomainKey( rtrim( $this->mServerName . '/' . $this->langCode, '/' ) ),
```
Comment (sqreek) [inline extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php:356]: why rtrim?

File: extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php [hunk 3]
```diff
@@ -307,17 +307,10 @@ public function execute() {
...
-					];
-				} else {
-					$where = [
-						'city_domains.city_id = city_list.city_id',
-						'city_domains.city_domain' => $this->mServerName
-					];
-				}
+				$where = [
+					'city_domains.city_id = city_list.city_id',
+					'city_domains.city_domain' => rtrim( $this->mServerName . '/' . $this->langCode, '/' )
```
Comment (sqreek) [inline extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php:312]: why rtrim?

File: extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php:255]: WF domain entries dont have a trailing slash so adding one here breaks domain cache invalidation where no slash is used

**Review by macbre — APPROVED** [review]

**Pull request merged**